nshlib/nsh_alias.c

```diff
@@ -0,0 +1,458 @@
+/****************************************************************************
+ * apps/nshlib/nsh_alias.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+
+#include <nuttx/queue.h>
+
+#include "nsh.h"
+#include "nsh_console.h"
+
+#ifdef CONFIG_NSH_ALIAS
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* State flags */
+
+#define ALIAS_INIT  0x01
+
+/* Macro to get head of alias list */
+
+#define alias_head(list)        (FAR struct nsh_alias_s *)sq_peek(list)
+#define alias_remfirst(list)    (FAR struct nsh_alias_s *)sq_remfirst(list)
+
+/* Initialize the alias shell container (if necessary) */
+
+#define do_alias_init(vtbl) \
+  do \
+    { \
+      if (!(((FAR struct console_stdio_s *)(vtbl))->flags & ALIAS_INIT)) \
```

Comment:
we can check the field in both alist and afreelist is null instead, and remove flags field and ALIAS_INIT macro.

Reply:
Sure, I can merge the check into alias_init.